cmd/oceantv/broadcast_manager.go

```diff
@@ -88,44 +88,58 @@ func (m *OceanBroadcastManager) CreateBroadcast(
 	store Store,
 	svc BroadcastService,
 ) error {
-	// We're going to add the date to the broadcast's name, so get this and format.
-	loc, err := time.LoadLocation(locationID)
+	// If the broadcast doesn't have an ID or is revoked or completed, create a new broadcast.
+	ctx := context.Background()
+	status, err := svc.BroadcastStatus(ctx, cfg.ID)
 	if err != nil {
-		return fmt.Errorf("could not load location: %w", err)
+		return fmt.Errorf("could not get broadcast status: %v", err)
 	}
+	if cfg.ID == "" || status == "" || status == broadcast.StatusRevoked || status == broadcast.StatusComplete {
```

Comment:
Could we check for the negative case instead i.e. conditions for using same broadcast. Then we don't need an else, and we don't need to nest this whole block e.g.
``` 
if cfg.ID != "" && status != "" && status != broadcast.StatusRevoked && status != broadcast.StatusComplete {
    err = m.Save ....
    // err check
    return nil 
}

// Create broadcast

Reply:
Also, should we have any sort of time based check i.e. if the start time is like yesterday then should we really use this broadcast ?

Reply:
The idea is that no uncompleted broadcasts from yesterday should exist to use.

Reply:
How can we be sure that assumption is true ?

Reply:
If it's not true, then that would be a bug that we should fix. I suppose it may be helpful to check it here too. I just prefer solving the problem at the source.

Reply:
Is it worth encapsulating the if test condition as a function, e.g., `IsValidBroadcast`?

Reply:
@trekhopton there's the possibility that we create a broadcast but can't get the hardware properly started or get the broadcast to transition to a live state (even after retries), so we'd be left with an unused broadcast. I don't think we currently have any mechanism that removes broadcasts that don't eventuate (we could have this though). We could check for these sorts of conditions (especially unexpected conditions) and log accordingly. 

Also, agree with @scruzin; we could put this logic in a IsValidBroadcast function :smile:

Reply:
Okay, I think we should remove broadcasts that don't eventuate or at least give the user that option. But yeah I think I agree it would be good to check that we're using a broadcast from today. I've added that now.

Also added the function.